nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/MoveAzureDataLakeStorage.java

```diff
@@ -0,0 +1,223 @@
+            if (!destinationDirectory.isEmpty() && ! sourceDirectory.isEmpty()) {
+                destinationPath = destinationDirectory + "/";
+            }
+
+            final DataLakeServiceClient storageClient = getStorageClient(context, flowFile);
+            final DataLakeFileSystemClient sourceFileSystemClient = storageClient.getFileSystemClient(sourceFileSystem);
+            final DataLakeDirectoryClient sourceDataLakeDirectoryClient = sourceFileSystemClient.getDirectoryClient(sourceDirectory);
+            final DataLakeFileSystemClient destinationFileSystemClient = storageClient.getFileSystemClient(destinationFileSystem);
+            final DataLakeDirectoryClient destinationDirectoryClient = destinationFileSystemClient.getDirectoryClient(destinationDirectory);
+            DataLakeFileClient fileClient = sourceDataLakeDirectoryClient.getFileClient(fileName);
+            final DataLakeRequestConditions sourceConditions = new DataLakeRequestConditions();
+            final DataLakeRequestConditions destConditions = new DataLakeRequestConditions();
+            final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+
+            try {
+                if (!destinationDirectory.isEmpty() && !destinationDirectoryClient.exists()) {
+                    destinationDirectoryClient.create();
+                }
+
+                if (!conflictResolution.equals(REPLACE_RESOLUTION)) {
+                    destConditions.setIfNoneMatch("*");
+                }
+
+                fileClient = fileClient.renameWithResponse(destinationFileSystem,
+                                destinationPath + fileName,
+                        sourceConditions,
+                        destConditions,
+                        null,
+                        null)
+                        .getValue();
+
+                final Map<String, String> attributes = new HashMap<>();
+                attributes.put(ATTR_NAME_SOURCE_FILESYSTEM, sourceFileSystem);
+                attributes.put(ATTR_NAME_SOURCE_DIRECTORY, sourceDirectory);
+                attributes.put(ATTR_NAME_FILESYSTEM, destinationFileSystem);
+                attributes.put(ATTR_NAME_DIRECTORY, destinationDirectory);
+                attributes.put(ATTR_NAME_FILENAME, fileName);
+                attributes.put(ATTR_NAME_PRIMARY_URI, fileClient.getFileUrl());
+                attributes.put(ATTR_NAME_LENGTH, String.valueOf(fileClient.getProperties().getFileSize()));
+                flowFile = session.putAllAttributes(flowFile, attributes);
+
+                session.transfer(flowFile, REL_SUCCESS);
+                final long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                session.getProvenanceReporter().send(flowFile, fileClient.getFileUrl(), transferMillis);
+            } catch (DataLakeStorageException dlsException) {
+                if (dlsException.getStatusCode() == 409) {
```

Comment:
Do you think the two condition branches can be unified like this?

`if (dlsException.getStatusCode() == 409 && conflictResolution.equals(IGNORE_RESOLUTION) {
...
} else throw dlsException`